src/fmpq.h

```diff
@@ -202,6 +205,10 @@ FMPQ_INLINE int fmpq_print(const fmpq_t x)
 {
     return fmpq_fprint(stdout, x);
 }
+#else
+int _fmpq_print(const fmpz_t num, const fmpz_t den);
+int fmpq_print(const fmpq_t x);
```

Comment:
Maybe just de-inline `fmpq_print` and `_fmpq_print` and their relatives everywhere? Then that's one less thing that is different based on `FLINT_HAVE_FILE`

Reply:
Yes, I think both me and Fredrik believe they should not be inlined. But that should be a different PR